src/main/java/com/google/sps/servlets/MatchServlet.java

```diff
@@ -39,8 +45,26 @@ public void testOnlySetContext(SessionContext sessionContext) {
     this.sessionContext = sessionContext;
   }
 
+
+  //method to override AbuseDetectionFeature with Mock FOR TESTING        
+  public void testOnlySetAbuseDetection(AbuseDetection abuseFeature) {
+    this.abuseDetectionFeature = abuseFeature;
+  }
+
+
+  /*
+  * This method relates to rate limiting problem by analyzing the time of the 
+  * request and limiting the amount of requests coming in. This is a problem
+  * because malicious users can overload requests to the backend, so we created
+  * this method to help limit the amount of requests coming to the backend.
+  */
+  public boolean analyzeTimeOfRequest(HttpServletRequest request) {
```

Comment:
1. Do you need the request argument?
2. Does it make sense to write a one line method?

Reply:
1. No I don't that's unneeded. 
2. No, I'll just add it to doGET

Reply:
Done